includes/Core/REST_API/REST_Routes.php

```diff
@@ -585,31 +585,41 @@ function( $dataset ) use ( $module ) {
 						'methods'  => WP_REST_Server::READABLE,
 						'callback' => function( WP_REST_Request $request ) {
 							$post_id = false;
-							$is_url  = filter_var( $request['query'], FILTER_VALIDATE_URL );
-							if ( $is_url ) {
-								$post_id = url_to_postid( $request['query'] );
+							$query   = rawurldecode( $request['query'] );
+
+							if ( filter_var( $query, FILTER_VALIDATE_URL ) ) {
+								// Translate public/alternate reference URLs to local if different.
+								$query_url = str_replace(
+									trailingslashit( $this->context->get_reference_site_url() ),
+									trailingslashit( home_url() ),
+									$query
+								);
+								$post_id = url_to_postid( $query_url );
```

Comment:
If this returns a non-successful response (i.e. 0), we should probably return early with an empty array.

Currently, it'd fall back to running a query which doesn't make sense because we're in the case for a URL.

Reply:
Updated with a cleaner solution that doesn't require additional nesting/conditional 👍